capa/main.py

```diff
@@ -224,7 +231,12 @@ def find_code_capabilities(
 def find_file_capabilities(ruleset: RuleSet, extractor: FeatureExtractor, function_features: FeatureSet):
     file_features = collections.defaultdict(set)  # type: FeatureSet
 
-    for feature, va in itertools.chain(extractor.extract_file_features(), extractor.extract_global_features()):
+    if isinstance(extractor, StaticFeatureExtractor):
+        extractor_: StaticFeatureExtractor = cast(StaticFeatureExtractor, extractor)
+    else:
+        extractor_: DynamicFeatureExtractor = cast(DynamicFeatureExtractor, extractor)
```

Comment:
when doing a type switch like this, i recommend using `if this:.. elif that:... else: raise ValueError` so that if we ever introduce another candidate type, this code quickly fails.
```suggestion
    elif isinstance(extractor, DynamicFeatureExtractor):
        extractor_: DynamicFeatureExtractor = cast(DynamicFeatureExtractor, extractor)
    else:
        raise ValueError(f"unexpected extractor type: {extractor.__class__.__name__}")
```